### PR TITLE
Adds ability to specify which certificate to use

### PR DESCRIPTION
**Pull request by almassapargali**
Now you can set which certificate to use though env variable `SIGH_CERTIFICATE`. If none of `SIGH_CERTIFICATE` or expiry date params set, it'll include all certificates on development profiles, and only first on production profiles.

For further consideration:
There are 2 options to specify certificate which are implemented, and 1 could be. To list them:
1. Expiry date: despite the fact that it was already [implemented](https://github.com/KrauseFx/sigh#advanced) people [kept](https://github.com/KrauseFx/sigh/issues/17) [asking](https://github.com/KrauseFx/sigh/issues/24) for it.
2. Certificate name (like `'Almas Sapargali'`): that's what I've implemented in this PR. It may be ambiguous sometimes, since we may have many certificates with same name. In this case all of them gets used.
3. Certificate ID aka 'the right way ©': while being the preferred way, I found it quite confusing, because DevCenter uses another ID for certificates, not what we see in Keychain Access.

I suggest to pick one of these and move with it.

Also, this PR includes a few Ruby refactoring.

### CODE REVIEW

File: lib/sigh/developer_center.rb [hunk 1]
```diff
@@ -320,7 +318,7 @@ def create_profile(app_identifier, type, cert_date)
     end
 
     def renew_profile(profile_id, type, cert_date)
-      certificate = code_signing_certificate(type, cert_date)
+      certificate = code_signing_certificates(type, cert_date).first
```
Comment (almassapargali) [inline lib/sigh/developer_center.rb]: Also, I don't like this. Though I'm not sure which page is this, where we have only radio inputs.

File: lib/sigh/developer_center.rb [hunk 2]
```diff
@@ -197,37 +192,27 @@ def maintain_app_certificate(app_identifier, type, force, cert_date)
...
-              next # that's an Ad Hoc profile. I didn't find a better way to detect if it's one ... skipping it
-            end
-            if type != APPSTORE and details['provisioningProfile']['deviceCount'] == 0
-              next # that's an App Store profile ... skipping it
-            end
+            # that's an Ad Hoc profile. I didn't find a better way to detect if it's one ... skipping it
+            next if type == APPSTORE and details['provisioningProfile']['deviceCount'] > 0
+
+            # that's an App Store profile ... skipping it
+            next if type != APPSTORE && details['provisioningProfile']['deviceCount'] == 0
```
Comment (almassapargali) [inline lib/sigh/developer_center.rb]: http://avdi.org/devblog/2010/08/02/using-and-and-or-in-ruby/

**Comment by almassapargali** [comment]
@alistra, @tychop, @Rusik could you please test this out? I did run couple of times. All seems OK.

File: lib/sigh/developer_center.rb [hunk 3]
```diff
@@ -197,37 +192,27 @@ def maintain_app_certificate(app_identifier, type, force, cert_date)
...
-              next # that's an Ad Hoc profile. I didn't find a better way to detect if it's one ... skipping it
-            end
-            if type != APPSTORE and details['provisioningProfile']['deviceCount'] == 0
-              next # that's an App Store profile ... skipping it
-            end
+            # that's an Ad Hoc profile. I didn't find a better way to detect if it's one ... skipping it
+            next if type == APPSTORE && details['provisioningProfile']['deviceCount'] > 0
+
+            # that's an App Store profile ... skipping it
+            next if type != APPSTORE && details['provisioningProfile']['deviceCount'] == 0
```
Comment (almassapargali) [inline lib/sigh/developer_center.rb]: http://avdi.org/devblog/2010/08/02/using-and-and-or-in-ruby/
Yes, @KrauseFx I'm aware you like 'and', sorry 💔

Comment (tychop) [inline lib/sigh/developer_center.rb]: I can still forcefully renew an enterprise profile. So no problems from this side.

**Comment by KrauseFx** [comment]
Sorry for being so quite the last few days, I'll test and merge everything this weekend :+1:

**Comment by KrauseFx** [comment]
Looks great, and is almost ready to merge. Please also update the README to explain the new option.

Also, as you've mentioned before, we should really soon move the shared code (e.g. login, update checker, and also things like profile selection) into a shared gem. 

There is already one shared gem (https://github.com/KrauseFx/CredentialsManager), which takes care of passwords and the Appfile.

**Comment by almassapargali** [comment]
Sure, will update README soon.

**Pull request merged**